projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -75,10 +112,21 @@ export class AddToCartComponent implements OnInit, OnDestroy {
   protected setStockInfo(product: Product): void {
     this.quantity = 1;
     this.hasStock = Boolean(
-      product.stock && product.stock.stockLevelStatus !== 'outOfStock'
+      product.stock && product.stock?.stockLevelStatus !== 'outOfStock'
     );
     if (this.hasStock && product.stock?.stockLevel) {
-      this.maxQuantity = product.stock.stockLevel;
+      this.maxQuantity = product.stock?.stockLevel;
+    } else {
+      this.maxQuantity = 0;
```

Comment:
do we need this? can't we just verify in the getInventory conditional statement to see if it's undefined, so this.maxQuantity should be good enough

Reply:
do we really need this?